internal/core/window.rs

```diff
@@ -205,6 +205,60 @@ struct PopupWindow {
     component: ComponentRc,
 }
 
+#[derive(Default)]
+struct ClickState {
+    click_count_time_stamp: Cell<Option<crate::animations::Instant>>,
+    click_count: Cell<u8>,
+}
+
+impl ClickState {
+    fn reset(&self) {
+        self.click_count.set(0);
+        self.click_count_time_stamp.set(Some(crate::animations::Instant::now()));
+    }
+    fn check_repeat(&self, mut mouse_event: MouseEvent) -> MouseEvent {
+        match mouse_event {
+            MouseEvent::Pressed { position, button, .. } => {
+                if let Some(click_count_time_stamp) = self.click_count_time_stamp.get() {
+                    if crate::animations::Instant::now() - click_count_time_stamp
+                        < crate::platform::PLATFORM_INSTANCE
+                            .with(|p| p.get().map(|p| p.click_interval()))
+                            .unwrap_or_default()
+                        && button == PointerEventButton::Left
+                    {
+                        self.click_count.set(self.click_count.get() + 1);
+                        self.click_count_time_stamp.set(Some(crate::animations::Instant::now()));
+                        mouse_event = MouseEvent::Pressed {
+                            position: position,
+                            button: button,
+                            click_count: self.click_count.get(),
+                        }
+                    } else {
+                        self.reset();
+                    }
+                } else {
+                    self.reset();
+                }
+            }
+            MouseEvent::Released { position, button, .. } => {
+                if self.click_count_time_stamp.get().is_none() {
+                    self.click_count_time_stamp.set(Some(crate::animations::Instant::now()));
```

Comment:
I don't think we should set this.

Otherwise:
 1. press
 2. wait long time
 3. release
 4. press quickly

will be undertstood as a double click.
IMHO, it the click timestamp should only be set on press.  (i'd call it press_time_stamp or something like that)

Reply:
I don't see press as double click with this, more like you have an extra field with the current internal click count. And you can choose what to do on release and press depending on click_count or ignore it.

Reply:
done